### DISCUSSION TITLE
Ajuda iniciante, código não funciona

### DISCUSSION THREAD

**Discussion by EdioZ** [frontendbr/forum#2275] [Perguntas e Respostas] [answered]
Bom dia pessoal, sou novo aqui e estou aprendendo Javascript bem básico, e me deparei com um erro que não estou conseguindo entender, bem simples mas atravancou meu caminho. Diz no console que nu1 não foi definida na função "somar", é isso mesmo? Para mim está tudo certo. 

Não sei se é melhor enviar por código ou imagem, mas na imagem abaixo está todo o código. Se alguém puder me ajudar agradeço. 

![problema-ecma-1](https://user-images.githubusercontent.com/104085857/164266640-b9fc89cf-a5e4-4b19-b546-7b04ad277420.jpg)

**Comment by felquis** [comment, marked as answer]
Fala @EdioZ tudo bem? Qual é seu nome?

Então, `nu1` no seu código de fato não foi definida, não há nenhuma const, let ou var definindo qual é o valor de nu1 e na linha 29 você tenta usa-la antes de definir.

### Como definir uma variável no JavaScript usando let

Você deve fazer assim:
```js
let nomeDaVariavel = 'Qualquer referencia de valor'
```

No seu printscreen aparece:
```js
let = nu1.document...
```

Isso me parece que tem um erro de sintaxe e não funciona.

Seu código correto ficaria assim:
```js
let nu1 = document.getElementById('text1')
let nu2 = document.getElementById('text2')
```

É isso...
Espero que isso te ajude!!

**Reply by EdioZ** [reply]
Caro @felquis meu nome é Edson, muito prazer. Claro, não acredito que não vi isso kkkk, na variável n1 eu coloquei certo antes de = e nu1 coloquei depois assim não ficou definida mesmo. Ajudou muito. obrigado, agora posso continuar em paz kkk. Abraço.
